cmd/tusd/cli/serve.go

```diff
@@ -92,7 +113,42 @@ func Serve() {
 		stdout.Printf("You can now upload files to: http://%s%s", address, basepath)
 	}
 
-	if err = http.Serve(listener, nil); err != nil {
-		stderr.Fatalf("Unable to serve: %s", err)
+	var g run.Group
+
+	g.Add(func() error {
+		if err = s.Serve(listener); err != nil {
```

Comment:
The Golang documentation says following:

> When Shutdown is called, Serve, ListenAndServe, and ListenAndServeTLS immediately return ErrServerClosed. Make sure the program doesn't exit and waits instead for Shutdown to return. 

So, I think we need a waiting mechanism here, don't we?

Reply:
> I interpret that as avoid calling os.Exit() and just call Shutdown() instead

I don't think that is right. We must make sure that the main goroutine does not end before the call to Shutdown() has completed. I am not sure if run.Group would guarantee that.

Reply:
My understanding is that the call to Shutdown aborts if it does not complete within the deadline associated from the supplied context.

The context deadline should likely be configured in accordance with the interval for an individual chunk request, so that restarting the server may happen with seemingly minimal interruption.